lib/bitarray/bit_array.rb

```diff
@@ -0,0 +1,93 @@
+class BitArray
+  attr_reader :field, :reverse_byte, :size
+  include Enumerable
+
+  VERSION = "1.4.0"
+  HEADER_LENGTH = 8 + 1 # QC (@size, @reverse_byte)
+
+  def initialize(size, field = nil, reverse_byte: true)
+    @size = size
+    @field = field || "\0" * (size / 8 + 1)
+    @reverse_byte = reverse_byte
+  end
+
+  # Set a bit (1/0)
+  def []=(position, value)
+    if value == 1
+      @field.setbyte(position >> 3, @field.getbyte(position >> 3) | (1 << (byte_position(position) % 8)))
+    else
+      @field.setbyte(position >> 3, @field.getbyte(position >> 3) & ~(1 << (byte_position(position) % 8)))
+    end
+  end
+
+  # Read a bit (1/0)
+  def [](position)
+    (@field.getbyte(position >> 3) & (1 << (byte_position(position) % 8))) > 0 ? 1 : 0
+  end
+
+  def ==(rhs)
+    @size == rhs.size && @reverse_byte == rhs.reverse_byte && @field == rhs.field
+  end
+
+  # Allows joining (union) two bitarrays of identical size.
+  # The resulting bitarray will contain any bit set in either constituent arrays.
+  # |= is implicitly defined, so you can do source_ba |= other_ba
+  def |(rhs)
+    raise ArgumentError.new("Bitarray sizes must be identical") if @size != rhs.size
+    raise ArgumentError.new("Reverse byte settings must be identical") if @reverse_byte != rhs.reverse_byte
+
+    combined = BitArray.new(@size, @field, reverse_byte: @reverse_byte)
+    rhs.field.each_byte.inject(0) do |byte_pos, byte|
+      combined.field.setbyte(byte_pos, combined.field.getbyte(byte_pos) | byte)
+      byte_pos + 1
+    end
+
+    combined
+  end
```

Comment:
I wrote this because it allows parallelising bloom filter creation. I don't personally need this method, but figured it might be useful for others.